File: bialydunajec-academic-ministry/presentation/src/main/kotlin/org/bialydunajec/academicministry/rest/v1/admin/request/CreateAcademicPriestRequest.kt
package org.bialydunajec.academicministry.rest.v1.admin.request

import org.bialydunajec.academicministry.application.dto.PersonalTitleDto
import org.bialydunajec.ddd.base.dto.ExtendedDescriptionDto

internal class CreateAcademicPriestRequest(
        val firstName: String,
        val lastName: String,
        val personalTitle: PersonalTitleDto?,
        val emailAddress: String?,
        val phoneNumber: String?,
        val description: ExtendedDescriptionDto?,
        val photoUrl: String?
)

File: frontend-adminpanel/projects/bialydunajec-admin/src/app/academic-ministry/service/rest/request/create-academic-priest.request.ts
import {ExtendedDescriptionDto} from '../../../../shared/service/rest/dto/extended-description.dto';
import {PersonalTitleDto} from '../dto/personal-title.dto';

export class CreateAcademicPriestRequest {
  firstName: string;
  lastName: string;
  personalTitle?: PersonalTitleDto;
  emailAddress?: string;
  phoneNumber?: string;
  description?: ExtendedDescriptionDto;
  photoUrl?: string;

  constructor(firstName: string, lastName: string, personalTitle: PersonalTitleDto, emailAddress: string, phoneNumber: string, description: ExtendedDescriptionDto, photoUrl: string) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.personalTitle = personalTitle;
    this.emailAddress = emailAddress;
    this.phoneNumber = phoneNumber;
    this.description = description;
    this.photoUrl = photoUrl;
  }
}
